presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/thrift/ThriftSqlFunctionExecutor.java

```diff
@@ -0,0 +1,80 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.execution.thrift;
+
+import com.facebook.drift.TException;
+import com.facebook.drift.client.DriftClient;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeSignature;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.SqlFunctionHandle;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.ThriftScalarFunctionImplementation;
+import com.facebook.presto.thrift.api.datatypes.PrestoThriftBlock;
+import com.facebook.presto.thrift.api.udf.ThriftFunctionHandle;
+import com.facebook.presto.thrift.api.udf.ThriftUdfService;
+import com.facebook.presto.thrift.api.udf.ThriftUdfServiceException;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.facebook.airlift.concurrent.MoreFutures.toCompletableFuture;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.facebook.presto.thrift.api.udf.ThriftUdfPage.thriftPage;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class ThriftSqlFunctionExecutor
+{
+    private final DriftClient<ThriftUdfService> thriftUdfClient;
+
+    @Inject
+    public ThriftSqlFunctionExecutor(DriftClient<ThriftUdfService> thriftUdfClient)
+    {
+        this.thriftUdfClient = thriftUdfClient;
+    }
+
+    public CompletableFuture<Block> executeFunction(ThriftScalarFunctionImplementation functionImplementation, Page input, List<Integer> channels, List<Type> argumentTypes, Type returnType)
+    {
+        if (thriftUdfClient == null) {
+            throw new UnsupportedOperationException("Thrift function execution is not supported");
+        }
+        ImmutableList.Builder<PrestoThriftBlock> blocks = ImmutableList.builder();
+        for (int i = 0; i < channels.size(); i++) {
+            Block block = input.getBlock(channels.get(i));
+            blocks.add(PrestoThriftBlock.fromBlock(block, argumentTypes.get(i)));
+        }
+        SqlFunctionHandle functionHandle = functionImplementation.getFunctionHandle();
+        SqlFunctionId functionId = functionImplementation.getFunctionHandle().getFunctionId();
+        try {
+            return toCompletableFuture(thriftUdfClient.get(Optional.of(functionImplementation.getLanguage().getLanguage())).invokeUdf(
+                    new ThriftFunctionHandle(
+                            functionId.getFunctionName().toString(),
+                            functionId.getArgumentTypes().stream()
```

Comment:
This comment doesn't request any change. Just purely a comment:  at first glance seeing "argument types" contained in `FunctionId` looks weird 😃 

I can totally imagine how does this happen. -- We are running out of words such as `FunctionSigature`, etc. ... So `FunctionId` becomes the best word. 😂

Reply:
Hmm now you mention it, it does feel non-obvious. But the unique identifier of a function is name + argument types so that's why the name is `FunctionId` and it includes the name and the argument types.